### PR TITLE
Adding CI job to validate that no custom patch is added to repo

### PR DESCRIPTION
**Pull request by davidferlay**

### CODE REVIEW

File: scripts/makefile/patchval.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env sh
+
+# Files addition to watch for
+PATCH_COUNT=$(git show --name-status | grep "^A" | grep -c "\.patch" )
```
Comment (andypost) [inline scripts/makefile/patchval.sh:4]: git show displays only last commit, for CI better to pass and previous commit to compare to

**Review by andypost — CHANGES_REQUESTED** [review]
needs work to allow look for a patch in list of commits

File: scripts/makefile/patchval.sh (hunk 1 above)
Comment (davidferlay) [reply scripts/makefile/patchval.sh:4]: Good catch, gonna dig that

Comment (davidferlay) [reply scripts/makefile/patchval.sh]: Fixed!

File: scripts/makefile/patchval.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  # ... when not in Gitlab CI
+  PARENT_BRANCH=$(git show-branch | sed "s/].*//" | grep "\*" | grep -v "$(git rev-parse --abbrev-ref HEAD)" | head -n1 | sed "s/^.*\[//")
+else
+  # ... when in Gitlab CI
+  PARENT_BRANCH=$CI_MERGE_REQUEST_TARGET_BRANCH_NAME
+fi
+
+# Diff current branch against parent branch and make validation
+git fetch -q origin "$PARENT_BRANCH"
+PATCH_FILE_COUNT=$(git diff origin/"$PARENT_BRANCH" --name-status --diff-filter=A | grep -c "\.patch")
```
[the remark this answers is not in the document]
Comment (gaydabura) [reply scripts/makefile/patchval.sh:15]: I guess you may parse composer.json without context in this case, as goal is to block local patche, nvm in which commit it was added

[the remark this answers is not in the document]
Comment (davidferlay) [reply scripts/makefile/patchval.sh:15]: Makes sense! Downside is that jq needed (in php container), so job will not be able to run without site (at least containers) built. Meaning job will need to be moved outside of "sniffers"

[the remark this answers is not in the document]
Comment (andypost) [reply scripts/makefile/patchval.sh:15]: In php we have php to parse json

[the remark this answers is not in the document]
Comment (davidferlay) [reply scripts/makefile/patchval.sh:15]: jq makes it usable even out of php projects : d

**Pull request merged**